features/Native Text Indices

```diff
@@ -0,0 +1,58 @@
+---
+description: This page talks about native text indices and corresponding search functionality in Pinot
+---
+
+# History Of Text Indexing And Search in Pinot
+Pinot supports text indexing and search by building Lucene indices as "sidecars" to the main Pinot segments.
+While this is a great technique, it essentially limits the avenues of optimizations that can be done for
+Pinot specific use cases of text search.
+
+# How Is Pinot Different?
+Pinot, or any other database/OLAP engine, do not need to conform to the entire full text search DSL that
+is traditionally used by FTS engines like ElasticSearch and Solr. Looking at traditional SQL like text
+search use cases, majority of text searches comprise of three patterns -- prefix wildcard queries, postfix
+wildcard queries and term queries.
+
+# Native Text Indices in Pinot
+Native text indices are built from the ground up. They use a custom text indexing engine, coupled with Pinot's
+powerful inverted indices, to provide a super fast text search experience.
+
+# Benefits of Native Text Indices
+Native text indices are 80-120% faster than Lucene based indices for the text search use cases mentioned above.
+They are also 40% smaller on disk.
+
+# Real Time Indexing And Search
+A new feature that native text indices support are real time text search. For REALTIME tables, native text
+indices allow data to be indexed in memory in the text index, while concurrently supporting text searches
+on the same index.
+
+Historically, most text indices depend on the in memory text index being written to first and then sealed,
+before searches are possible. This limits the freshness of the search, being near real time at best.
+
+Native text indices come with a custom in memory text index, which allows for real time indexing and search.
+
+# Searching Native Text Indices
```

Comment:
Would be good to call out any syntactic/semantic differences from existing text indices, and what's supported what's not.

Reply:
Highlighted the difference in TEXT_CONTAINS details.